plugins/modules/network/illumos/dladm_vnic.py

```diff
@@ -176,22 +181,22 @@ def is_valid_unicast_mac(self):
 
         mac_re = re.match(self.UNICAST_MAC_REGEX, self.mac)
 
-        return mac_re is None
+        return mac_re is not None
 
     def is_valid_vlan_id(self):
 
-        return 0 <= self.vlan <= 4095
+        return 0 < self.vlan <= 4095
```

Comment:
This is the only change I cannot really judge and would wait for a maintainer to look at - except there seem to be no maintainers. Do you have a link to some docs which say that 0 is not a valid VLAN ID?

Reply:
I also had the upper range wrong and fixed to exclude 4095 as well in f1a95c8.

Here are some reasons that it's *now* correct:
* Trying to create a vnic with a VID of 0/4095 on my physical Illumos host (OmniOS) emits an error about it being unacceptable:
    ```
    sudo dladm create-vnic -t -l net0 -v 0 test0
    dladm: invalid VLAN identifier '0'

    sudo dladm create-vnic -t -l net0 -v 4095 test0
    dladm: invalid VLAN identifier '4095'
    ```
* The Illumos source code for [`dladm-vnic`](https://github.com/illumos/illumos-gate/blob/master/usr/src/cmd/dladm/dladm.c#L4716-L4717) confirms this logic and behavior.
* A VID of 0 is a reserved part of [802.1Q](https://en.wikipedia.org/wiki/IEEE_802.1Q); you'll typically only see 0-tagged frames in the context of [802.1p](https://en.wikipedia.org/wiki/IEEE_P802.1p).